pixeltable/iterators/document.py

```diff
@@ -442,21 +446,14 @@ def _emit_text() -> str:
             accumulated_text.clear()
             return txt
 
-        for page_idx, page in enumerate(doc.pages()):
+        for page_idx, page in enumerate(doc):
             img: PIL.Image.Image | None = None
             if Element.IMAGE in self._elements:
-                pix = page.get_pixmap(dpi=self._image_dpi)
-                img = PIL.Image.open(io.BytesIO(pix.tobytes(self._image_format)))
-
-            for block in page.get_text('blocks'):
-                x1, y1, x2, y2, text, *_ = block
-                _add_cleaned(text)
-                if accumulated_text and emit_on_paragraph:
-                    bbox = {'x1': x1, 'y1': y1, 'x2': x2, 'y2': y2}
-                    md = DocumentSectionMetadata(page=page_idx, bounding_box=bbox)
-                    yield DocumentSection(text=_emit_text(), metadata=md)
-
-            if accumulated_text and emit_on_page and not emit_on_paragraph:
+                img = page.render().to_pil()
```

Comment:
a bit more concise
img = page.render().to_pil() if element.image in self._elements else None